README.md

```diff
@@ -0,0 +1,94 @@
+# switchtec-dma
+
+Switchtec Gen4 PCIe switch introduces a new hardware DMA engine to offload
+expensive memory operations, such as large copies, from host processors. A
+complete solution of the Switchtec DMA involves not only the hardware DMA
+engine, but also embedded firmware and host software.
+
+The Switchtec DMA kernel module exports the access to the Switchtec DMA engine
+to the upper layer host software (DMA client).
+
+## Interface
+
+When configured, the Switchtec devices advertise a special DMA endpoint which
+enables the DMA functionality. BAR0 of the endpoint exports the firmware and
+hardware registers for the driver to configure and control the DMA engine.
+
+The switchtec dma driver maintains two queues in host memory space, a submission
+queue (SQ) and a completion queue (CQ), to communicate with the hardware DMA
+engine. The Switchtec DMA driver adds Submission Elements to the SQ on behalf of
+the DMA client when a DMA transfer is kicked off. The DMA engine adds Completion
+Elements to the CQ when a DMA transfer has finished.
```

Comment:
1) append CE after Completion Elements.
2) is the CE added by device/hardware instead of DMA engine?

Reply:
For 1. Will update. 
For 2. DMA engine is the DMAC/hardware. I think DMA engine is a common way to refer to this specific hardware piece (DMAC).